### ISSUE TITLE
[Outreachy Task Submission] Add Keyboard Support For Add Post Modal

### ISSUE THREAD

**Issue by Aquilaafuadajo** [ushahidi/platform#4897]
**Description**

Currently, the Add Post Dialog is not accessible using keyboard and screen readers. The following are the accessibility issues identified with this section:

1. Survey Items are not pronounced properly
2. Survey Items do not receive focus when tabbing
3. Survey items do not respond to click events
